src/app/catalog/header/header.component.ts

```diff
@@ -0,0 +1,20 @@
+import { Component, OnInit } from '@angular/core';
+import { LocalstorageService } from 'src/app/services/localstorage.service';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss'],
+})
+export class HeaderComponent implements OnInit {
+  userName: string = '';
+  constructor(private localStorageService: LocalstorageService) {}
+
+  ngOnInit(): void {
+    this.getUserName();
+  }
+
+  getUserName() {
```

Comment:
redundant due to the fact that it should exist in AuthService -> so you just need to inject that service and display it inside template via async pipe